apitally/litestar.py

```diff
@@ -0,0 +1,176 @@
+import json
+import sys
+import time
+from importlib.metadata import version
+from typing import Callable, Dict, List, Optional
+
+from litestar.app import DEFAULT_OPENAPI_CONFIG, Litestar
+from litestar.config.app import AppConfig
+from litestar.connection import Request
+from litestar.datastructures import Headers
+from litestar.enums import ScopeType
+from litestar.plugins import InitPluginProtocol
+from litestar.types import ASGIApp, Message, Receive, Scope, Send
+
+from apitally.client.asyncio import ApitallyClient
+
+
+__all__ = ["ApitallyPlugin"]
+
+
+class ApitallyPlugin(InitPluginProtocol):
+    def __init__(
+        self,
+        client_id: str,
+        env: str = "dev",
+        app_version: Optional[str] = None,
+        filter_openapi_paths: bool = True,
+        identify_consumer_callback: Optional[Callable[[Request], Optional[str]]] = None,
+    ) -> None:
+        self.client: ApitallyClient = ApitallyClient(client_id=client_id, env=env)
```

Comment:
```suggestion
        self.client = ApitallyClient(client_id=client_id, env=env)
```
Is mypy angry? seems like it could infer but maybe wrong

Reply:
Mypy thinks the type is `ApitallyClientBase` because of https://github.com/apitally/python-client/blob/0a76d70aee80ff4051afeeb5745b483aca82a317/apitally/client/base.py#L33.

I was going to try and improve the typing there, but forgot. Thanks for the reminder!